modules/date-localizer/main/index.coffee

```diff
@@ -0,0 +1,168 @@
+class DateLocalizer
+  zeroPad = hx.format.zeroPad(2)
+
+  # get the display order for the date so dates can be displayed correctly when localised
+  dateOrder: -> ['DD','MM','YYYY']
+
+  # get the day the week starts on, 0 for sunday, 1 for monday etc.
+  weekStart: -> 0
+
+  # localise the days of the week and return as array of 2 char days ('Su', 'Mo' etc.)
+  weekDays: -> ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa']
+
+  # localise 'today' text
+  todayText: -> 'Today'
+
+  # localise the day of the month and optionally zero pad (01, 02)
+  day: (day, pad) -> if pad then zeroPad(day) else day
+
+  # localise the month in the format of mmm (Jan, Feb etc.) or 01, 02 etc.
+  month: (month, short) ->
+    if short then zeroPad(month + 1)
+    else ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec'][month]
+
+  # localise the full year in the format of yyyy
+  year: (year) -> year
+
+  # localise a date object to return a date string of dd/mm/yyyy (or localised format)
+  date: (date, useInbuilt) ->
+    if useInbuilt
+      date.getFullYear() + '-' +
+      zeroPad(date.getMonth() + 1 )+ '-' +
+      zeroPad(date.getDate())
+    else
+      zeroPad(date.getDate()) + '/' + zeroPad(date.getMonth() + 1) + '/' + date.getFullYear()
+
+  # localise a date object to return a time string of hh:mm or hh:mm:ss (or localised format)
+  time: (date, showSeconds) ->
+    date = hx.preferences.applyTimezoneOffset(date)
+    timeString = date.getHours() + ':' + zeroPad date.getMinutes()
+    if showSeconds
+      timeString += ':' + zeroPad date.getSeconds()
+    timeString
+
+  # check if a time is a valid time (time as array of [hh, mm, ss])
+  checkTime: (time) ->
+    not isNaN(time[0]) and not isNaN(time[1]) and not isNaN(time[2])
+
+  # convert a localised date string back to a date object (unlocalise)
+  stringToDate: (dateString, useInbuilt) ->
+    if useInbuilt
+      order = ['YYYY', 'MM', 'DD']
+      split = dateString.split('-')
+    else
+      order = @dateOrder()
+      split = dateString.split('/')
+    allValid = split.length is 3 and not split.some (e) -> e is '' or e is '0'
+    if allValid
+      format = ''
+      for w, i in order
```

Comment:
or:

    hx.zip [w, split]
      .map ([w, splitElemStr]) ->
        splitElem = Number splitElemStr
        switch w
          when 'DD'
            daysValid = part.length < 3 and part isnt ''
            day = splitElem
       # etc